components/soap/src/main/java/org/switchyard/component/soap/composer/SOAPContextMapper.java

```diff
@@ -90,11 +102,48 @@ public void mapFrom(SOAPBindingData source, Context context) throws Exception {
         if (source.getStatus() != null) {
             context.setProperty(HTTP_RESPONSE_STATUS, source.getStatus()).addLabels(SOAP_MIME_LABELS);
         }
+        
+        //Gets the propagation label from the service domain
+        String propagationLabel=null;
+        if(context.getProperty(SERVICE_REFERENCE_PROPERTY)!=null){
+            ServiceDomain domain= ((ServiceReference)context.getProperty(SERVICE_REFERENCE_PROPERTY).getValue()).getDomain();
+            if(domain.getProperty(PropertyConstants.DOMAIN_PROPERTY_PROPAGATE_LABEL)!=null){
+            	propagationLabel=(String)domain.getProperty(PropertyConstants.DOMAIN_PROPERTY_PROPAGATE_LABEL);
+            }
+            
+        }
+
+      	
+
+   
         for (String key : source.getHttpHeaders().keySet()) {
             if (matches(key)) {
                 List<String> values = source.getHttpHeaders().get(key);
                 if (values != null) {
-                    context.setProperty(key, values).addLabels(SOAP_MIME_LABELS);
+                    //If the propagate header has ben set on the header
+                    if(key.startsWith(SOAP_PREFIX_PROPAGATE_HEADER)){
```

Comment:
IMO we shouldn't do this. Once the interaction gets out of SwitchYard world via reference binding, we don't have to guarantee the context properties are carried over.